website/server/controllers/api-v3/tasks.js

```diff
@@ -581,12 +581,18 @@ api.scoreTask = {
     // TODO move to common code?
     if (task.type === 'todo') {
       if (!wasCompleted && task.completed) {
-        removeFromArray(user.tasksOrder.todos, task._id);
+        await user.update({
+          $pull: { 'tasksOrder.todos': task._id }
+        }).exec();
       } else if (wasCompleted && !task.completed) {
-        let hasTask = removeFromArray(user.tasksOrder.todos, task._id);
-        if (!hasTask) {
-          user.tasksOrder.todos.push(task._id);
-        } // If for some reason it hadn't been removed previously don't do anything
+        // @TOOD: Why do we pull and push
+        // let hasTask = removeFromArray(user.tasksOrder.todos, task._id);
+        await user.update({
```

Comment:
we're still saving the user below

Reply:
Yea, I figured I'd need to remove the save's from most of it. I posted this to make sure I am heading in the write direction. I didn't realize using update skips much of the mongoose features.

Reply:
Hmm. I actually don't know if I can prevent this right now. This function probably needs to good refactoring to see.